HyperVExtension/src/HyperVExtension/Services/ArchiveProviderFactory.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using HyperVExtension.Models.VirtualMachineCreation;
+
+namespace HyperVExtension.Services;
+
+/// <summary>
+/// Provides a factory for creating archive providers based on the archive file extension.
+/// </summary>
+public sealed class ArchiveProviderFactory : IArchiveProviderFactory
+{
+    public IArchiveProvider CreateArchiveProvider(string extension)
```

Comment:
NIT: Add comment notifying users of `ArchiveProviderFactory` that only `.zip` is supported.

Reply:
As we add support for more I can add a comment in. So far I believe only Hart has plans to add another one, but not sure when that work would take place.